test/Microsoft.Azure.WebJobs.Host.UnitTests/Indexers/FunctionIndexerTests.cs

```diff
@@ -115,6 +115,22 @@ public void IndexMethod_IfMethodReturnsTask_DoesNotThrow()
         }
 
         [Fact]
+        public void IndexMethod_IfMethodReturnsAsyncVoid_Throws()
+        {
+            // Arrange
+            IFunctionIndexCollector index = CreateDummyFunctionIndex();
```

Comment:
See my expanded test - I need to pass in a mock trace writer. I'm going to just leave this test using the same tools all the other tests in the suite are using for now. If we want to we can convert all of them over later.